modules/internal/DefaultRectangular.chpl

```diff
@@ -1791,6 +1791,10 @@ module DefaultRectangular {
   }
 
   proc chpl_serialReadWriteRectangular(f, arr, dom) {
+    chpl_serialReadWriteRectangularHelper(f, arr, dom);
+  }
+
+  proc chpl_serialReadWriteRectangularHelper(f, arr, dom) {
```

Comment:
I basically want to print out ArrayViews over a ReplicatedArr like this:
```
for loc in targetLocales do on loc {
  writeln(loc);
  chpl_serialReadWriteRectangular(f, arr, dom); // pass in the ArrayView 'arr'
}
```

I cannot catch this case without inspecting the type of 'arr' ,and I cannot print it out without passing 'arr' down to chpl_serialReadWriteRectangular. To get around the recursion issue, I created the helper.